lib/util/itinerary.js

```diff
@@ -245,31 +245,133 @@ export function getLegBounds (leg) {
-  if (a.sortOrder !== b.sortOrder) {
-    aComp = a.sortOrder
-    bComp = b.sortOrder
-  } else if (!isNaN(parseInt(a.shortName)) && !isNaN(parseInt(b.shortName))) {
-    // Otherwise, if both short names can be parsed as integers, use these
-    // numbers for sorting.
-    aComp = parseInt(a.shortName)
-    bComp = parseInt(b.shortName)
-  } else {
-    // Otherwise, default to short or long name strings.
-    aComp = a.shortName || a.longName
-    bComp = b.shortName || b.longName
+/**
+ * Checks whether an appropriate comparison of numeric values can be made for
+ * sorting purposes. If both values are not valid numbers according to the
+ * isNaN check, then this function returns undefined which indicates that a
+ * secondary sorting criteria should be used instead. If one value is valid and
+ * the other is not, then the valid value will be given sorting priority. If
+ * both values are valid numbers, the difference is obtained as the sort value.
+ *
+ * An optional argument can be provided which will be used to obtain the
+ * comparison value from the comparison function arguments.
+ *
+ * IMPORTANT: the comparison values must be numeric values or at least be
+ * attempted to be converted to numeric values! If one of the arguments is
+ * something crazy like an empty string, unexpected behavior will occur because
+ * JavaScript.
+ *
+ * @param  {function} [objGetterFn] A function to obtain the comparison value
+ *  from the comparator function arguments
+ */
+function makeNumericValueComparator (objGetterFn) {
+  return (a, b) => {
+    let aVal
+    let bVal
+    if (typeof objGetterFn === 'function') {
+      aVal = objGetterFn(a)
+      bVal = objGetterFn(b)
+    } else {
+      aVal = a
+      bVal = b
+    }
+    // if both values aren't valid numbers, use the next sort criteria
+    if (isNaN(aVal) && isNaN(bVal)) return 0
+    // b is a valid number, b gets priority
+    if (isNaN(aVal)) return 1
+    // a is a valid number, a gets priority
+    if (isNaN(bVal)) return -1
+    // a and b are valid numbers, return the sort value
+    return aVal - bVal
+  }
+}
+
+/**
+ * Create a comparator function that compares string values. The comparison
+ * values feed to the sort comparator function are assumed to be objects that
+ * will have either undefined, null or string values at the given key. If one
+ * object has undefined, null or an empty string, but the other does have a
+ * string with length > 0, then that string will get priority.
+ *
+ * @param  {string} key The key from which to obtain the sort comparison value
+ */
+function makeObjectStringValueComparator (key) {
+  return (a, b) => {
+    const aVal = a[key]
+    const bVal = b[key]
+    // both a and b are uncomparable strings, return equivalent value
+    if (!aVal && !bVal) return 0
+    // a is not a comparable string, b gets priority
+    if (!aVal) return 1
+    // b is not a comparable string, a gets priority
+    if (!bVal) return -1
```

Comment:
Again, wondering about this embedded logic similar to https://github.com/opentripplanner/otp-react-redux/pull/124/files#r372464064